server/src/main/java/org/elasticsearch/rest/MethodHandlers.java

```diff
@@ -31,23 +31,25 @@
 final class MethodHandlers {
 
     private final String path;
-    private final Map<RestRequest.Method, RestHandler> methodHandlers;
+    private final Map<RestRequest.Method, Map<Version, RestHandler>> methodHandlers;
 
-    MethodHandlers(String path, RestHandler handler, RestRequest.Method... methods) {
+    MethodHandlers(String path, RestHandler handler, Version version,  RestRequest.Method... methods) {
```

Comment:
Can you a few unit tests for this class ?

Reply:
I also realised that changing the method signature was not necessary.